sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -63,6 +66,12 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
     private tokenCache: AccessTokenCache
   ) {
     super(nextPolicy, options);
+    const requiredMillisecondsBeforeNewRefresh = 30000;
```

Comment:
Commenting so we don't forget it - probably don't want to hardcode the delay like this. At the very least, we should pick the buffer, make it a const, and document it.

Reply:
> we should pick the buffer, make it a const, and document it

What do you mean by picking the buffer?

Reply:
You're hardcoding 30000 here. What does it mean? Why 30000?

Reply:
What do you think of this as a property of the BearerTokenAuthenticationPolicy class? Like:

```ts
export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
  // The automated token refresh will only start to happen at the
  // expiration date minus the value of millisecondsBeforeNewRefresh,
  // which is by default 30 seconds.
  public millisecondsBeforeNewRefresh: number = 30000;
```